### ISSUE TITLE
data2bids does not yet support handling of EMG data

### ISSUE THREAD

**Issue by annemonelies** [fieldtrip/fieldtrip#1196]

**Comment by robertoostenveld** [comment]
Just for the record: in principle it should not yet support EMG, since that is not part of the current BIDS specification. But as we discussed, it would be convenient to add something for EMG, not only to replicate the EEG, but also to think about metadata for electrode placement. and reference schemes. 

I propose that the function gives a warning in case cfg.dataype='emg', explaining that it is not part of the BIDS spec. We could also automatically add the EMG directories to the `.bidsignore` file, or at least document that as a possibility.

**Comment by robertoostenveld** [comment]
this might also be interesting for @marcelzwiers to follow.

**Comment by robertoostenveld** [comment]
I took a T1w and T2w of a random person, and some physiology recorded in the MRI scanner from another random person, and tested with the following script.

```
%%
% the first part is general to the different recordings

cfg = [];
cfg.bidsroot = 'bids';
cfg.sub = '139075';
cfg.ses = 'mri'; % all data was recorded in a single MRI session

cfg.InstitutionName             = 'Radboud University';
cfg.InstitutionalDepartmentName = 'Donders Institute for Brain, Cognition and Behaviour';
cfg.InstitutionAddress          = 'Kapittelweg 29, 6525 EN, Nijmegen, The Netherlands';

% REQUIRED
cfg.dataset_description.Name                = 'n/a';
cfg.dataset_description.BIDSVersion         = '1.2';
% OPTIONAL
cfg.dataset_description.License             = 'n/a';
cfg.dataset_description.Authors             = 'n/a';
cfg.dataset_description.Acknowledgements    = 'n/a';
cfg.dataset_description.Funding             = 'n/a';
cfg.dataset_description.ReferencesAndLinks  = 'n/a';
cfg.dataset_description.DatasetDOI          = 'n/a';

%%
% convert the T1w dicom files

cfg.dataset = t1wfile;
cfg.datatype = 'T1w';
data2bids(cfg);

%%
% convert the T2w dicom files

cfg.dataset = t2wfile;
cfg.datatype = 'T2w';
data2bids(cfg);

%%
% copy the EMG

% these are general fields and need to be added for EMG
% for MRI they can (hopefully) be read from the DICOM headers
cfg.Manufacturers = 'BrainProducts';
cfg.ManufacturersModelName = 'BrainAmp MR plus';
cfg.DeviceSerialNumber = 'n/a';
cfg.SoftwareVersions = 'n/a';

% FIXME we should agree upon the metadata fields for the JSON
% cfg.emg.SamplingFrequency         % this will be determined from the file
% cfg.emg.RecordingDuration         % this will be determined from the file, in seconds
% cfg.emg.RecordingType             % this will probably be determined from the file, continuous
% cfg.emg.PowerLineFrequency = 50;
% cfg.emg.HardwareFilters
% cfg.emg.SoftwareFilters
% cfg.emg.EMGChannelCount
% cfg.emg.EOGChannelCount
% cfg.emg.ECGChannelCount
% cfg.emg.ElectrodeManufacturer
% cfg.emg.ElectrodeManufacturersModelName
% cfg.emg.EMGPlacementScheme
% cfg.emg.EMGGround

cfg.method = 'copy'; % no need to convert
cfg.dataset = emgfile;
cfg.datatype = 'eeg'; % FIXME this should be emg

data2bids(cfg);
```

**Comment by robertoostenveld** [comment]
@annemonelies please check the changes in PR #1197. The updated test script now specifies `cfg.datatype='emg'` plus some EMG specific metadata fields. It results in the following data organization

```
./dataset_description.json
./sub-139075
./sub-139075/ses-mri
./sub-139075/ses-mri/emg
./sub-139075/ses-mri/emg/sub-139075_ses-mri_emg.json
./sub-139075/ses-mri/emg/sub-139075_ses-mri_emg.eeg
./sub-139075/ses-mri/emg/sub-139075_ses-mri_events.tsv
./sub-139075/ses-mri/emg/sub-139075_ses-mri_channels.tsv
./sub-139075/ses-mri/emg/sub-139075_ses-mri_emg.vmrk
./sub-139075/ses-mri/emg/sub-139075_ses-mri_emg.vhdr
./sub-139075/ses-mri/sub-139075_ses-mri_scans.tsv
./sub-139075/ses-mri/anat
./sub-139075/ses-mri/anat/sub-139075_ses-mri_T2w.json
./sub-139075/ses-mri/anat/sub-139075_ses-mri_T2w.nii
./sub-139075/ses-mri/anat/sub-139075_ses-mri_T1w.json
./sub-139075/ses-mri/anat/sub-139075_ses-mri_T1w.nii
./participants.tsv
```

where the _emg.json file contains

```
{
	"DeviceSerialNumber": "n/a",
	"ECGChannelCount": 1,
	"EMGChannelCount": 2,
	"EOGChannelCount": 0,
	"ElectrodeManufacturer": "Kendall",
	"ElectrodeManufacturersModelName": "schuim, 24mm",
	"EpochLength": 720.24,
	"InstitutionAddress": "Kapittelweg 29, 6525 EN, Nijmegen, The Netherlands",
	"InstitutionName": "Radboud University",
	"InstitutionalDepartmentName": "Donders Institute for Brain, Cognition and Behaviour",
	"Manufacturers": "BrainProducts",
	"ManufacturersModelName": "BrainAmp MR plus",
	"PowerLineFrequency": 50,
	"RecordingDuration": 720.24,
	"SamplingFrequency": 5000,
	"SoftwareVersions": "n/a"
}
```

Does this look meaningful? I realize that the task information is still missing, but that would be standard BIDS, not specific to EMG.

As we discussed in person: the json file should also document where the electrodes were placed (ref, ground, bipolar or not, etc). How do you propose to format that? You might want to have a look at https://bids-specification.readthedocs.io/en/stable/04-modality-specific-files/04-intracranial-electroencephalography.html#sidecar-json-_ieegjson for inspiration. The iEEG is the better example, since it needs to be more flexible than EEG, and certainly than MEG.

**Comment by robertoostenveld** [comment]
this now has been implemented with #1204. The functionality is not yet documented in the help of data2bids, but you can find it in the code.
